service/round.go

```diff
@@ -87,7 +88,17 @@ func (r *round) execute() error {
 	return nil
 }
 
-func (r *round) membershipProof(c []byte) ([][]byte, error) {
+func (r *round) membershipProof(c []byte, wait bool) ([][]byte, error) {
+	if wait {
+		<-r.closedChan
+	} else {
+		select {
+		case <-r.closedChan:
```

Comment:
why read from channel here if you have a default statement?

Reply:
to check whether `closedChan` was closed. if so, then it will proceed, instead of the default case, which basically checks if it wasn't closed.